src/App.tsx

```diff
@@ -1,29 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
+// import { event } from 'cypress/types/jquery';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const DEFAULT_PAGE = 1;
+const items: string[] = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(DEFAULT_PAGE);
+
+  const startItem = (currentPage - 1) * itemsPerPage;
+  const endItem =
+    startItem + itemsPerPage > items.length
+      ? items.length
+      : startItem + itemsPerPage;
+
+  const handlerPageChange = (value: number) => {
+    setCurrentPage(value);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${startItem + 1} - ${endItem} of 42)`}
```

Comment:
The total number of items (`42`) is hardcoded in the message. This should ideally reference the length of the `items` array to remain accurate if the number of items changes.